mesheryctl/internal/cli/root/filter/list.go

```diff
@@ -83,66 +86,106 @@ mesheryctl filter list 'Test Filter' (maximum 25 filters)
-				return nil
+		// Initialize pagination variables
+		startIndex := 0
+		endIndex := min(pageSize, len(data))
+		totalFilters := int(response.TotalCount)
+
+		for {
+			// Clear the terminal screen and print pagination details
+			if terminal.IsTerminal(int(os.Stdout.Fd())) {
+				fmt.Print("\033[H\033[2J") // ANSI escape code to clear the screen
 			}
+			fmt.Printf("Total number of filters: %d\n", totalFilters)
+			fmt.Printf("Page: %d of %d\n", (startIndex/pageSize)+1, (totalFilters+pageSize-1)/pageSize)
 
-			for _, v := range response.Filters {
-				FilterID := utils.TruncateID(v.ID.String())
-				var UserID string
-				if v.UserID != nil {
-					UserID = *v.UserID
-				} else {
-					UserID = "null"
-				}
-				FilterName := v.Name
-				CreatedAt := fmt.Sprintf("%d-%d-%d %d:%d:%d", int(v.CreatedAt.Month()), v.CreatedAt.Day(), v.CreatedAt.Year(), v.CreatedAt.Hour(), v.CreatedAt.Minute(), v.CreatedAt.Second())
-				UpdatedAt := fmt.Sprintf("%d-%d-%d %d:%d:%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year(), v.UpdatedAt.Hour(), v.UpdatedAt.Minute(), v.UpdatedAt.Second())
-				data = append(data, []string{FilterID, UserID, FilterName, CreatedAt, UpdatedAt})
+			// Display the current page of filters
+			displayCurrentPage(data[startIndex:endIndex], verbose)
+
+			// Check if we've displayed all filters
+			if endIndex == totalFilters {
+				break
 			}
-			utils.PrintToTableWithFooter([]string{"FILTER ID", "USER ID", "NAME", "CREATED", "UPDATED"}, data, []string{"Total", fmt.Sprintf("%d", response.TotalCount), "", "", ""})
-			return nil
-		}
 
-		// Check if meshery provider is set
-		if provider == "None" {
-			for _, v := range response.Filters {
-				FilterName := strings.Trim(v.Name, filepath.Ext(v.Name))
-				FilterID := utils.TruncateID(v.ID.String())
-				CreatedAt := fmt.Sprintf("%d-%d-%d", int(v.CreatedAt.Month()), v.CreatedAt.Day(), v.CreatedAt.Year())
-				UpdatedAt := fmt.Sprintf("%d-%d-%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year())
-				data = append(data, []string{FilterID, FilterName, CreatedAt, UpdatedAt})
+			// Wait for user input to navigate pages
+			fmt.Println("Press Enter or ↓ to continue, Esc or Ctrl+C to exit")
+
+			char, key, err := keyboard.GetKey()
+			if err != nil {
+				utils.Log.Error(err)
+				return err
 			}
-			utils.PrintToTableWithFooter([]string{"FILTER ID", "NAME", "CREATED", "UPDATED"}, data, []string{"Total", fmt.Sprintf("%d", response.TotalCount), "", ""})
-			return nil
-		}
-		for _, v := range response.Filters {
-			FilterID := utils.TruncateID(v.ID.String())
-			var UserID string
-			if v.UserID != nil {
-				UserID = utils.TruncateID(*v.UserID)
-			} else {
-				UserID = "null"
+
+			// Handle user input
+			if key == keyboard.KeyEsc || key == keyboard.KeyCtrlC {
+				break
+			} else if char == '\r' || key == keyboard.KeyArrowDown {
+				startIndex += pageSize
```

Comment:
We can improve the logic to check how many pages will be there. 
`total_count / page_size `. 
What do you think?